tests/helpers.py

```diff
@@ -17,34 +17,48 @@ def run_subprocess(command, working_dir='.', env=None, expected_returncode=0, in
     Helper function to run a shell command and report success/failure
     depending on the exit status of the shell command.
     """
+    env_ = os.environ.copy()
     if env is not None:
-        env_ = os.environ.copy()
         env_.update(env)
-        env = env_
+    env = env_
+
+    # Detect if we're running in CircleCI.
+    # https://circleci.com/docs/2.0/env-vars/
+    circleci = 'CIRCLECI' in env
+
+    # CircleCI has its own timeout mechanism. Disable ours.
+    timeout = None if circleci else 10*60
```

Comment:
Should we really have/permit sub-tests taking 10 minutes?

Reply:
Frankly we shouldn't be using the subprocess module's timeout mechanism at all. We have a clear problem with CircleCI sending jobs to the background for long enough that the timeout expires, which is why this PR disables the timeout in CircleCI. But other test environments are also likely to send jobs to the background.

@xvzcf what was the reason for switching from subprocess.run to Popen + communicate in dd739fc8

Reply:
This rings a bell, was it https://github.com/open-quantum-safe/liboqs/issues/852 ?